src/bson.ts

```diff
@@ -85,13 +88,12 @@ export const EJSON = {
   deserialize: EJSON_deserialize
 };
 
-export interface Document {
+export interface BSONDocument {
```

Comment:
Since there are DOM types here Document is the type of the global `document` object. We can rename this in the driver import but it seems best that this library doesn't clash.